DirectProgramming/DPC++FPGA/ReferenceDesigns/merge_sort/src/Misc.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef __MISC_HPP__
+#define __MISC_HPP__
+
+template <typename T>
+static constexpr T Pow2(T n) {
+  static_assert(std::is_integral<T>::value);
+  static_assert(std::is_unsigned<T>::value);
+  return T(1) << n;
+}
+
+template<typename T>
+constexpr bool IsPow2(T n) {
+  static_assert(std::is_integral<T>::value);
+  static_assert(std::is_unsigned<T>::value);
+  return (n != 0) && ((n & (n - 1)) == 0);
+}
+
+template <typename T>
+constexpr T Log2(T n) {
+  static_assert(std::is_integral<T>::value);
+  static_assert(std::is_unsigned<T>::value);
+  return ((n < 2) ? T(0) : T(1) + Log2(n / 2));

```

Comment:
Recursive functions are not allowed in kernels.  Doesn't really matter since you're not using this in a kernel, but why not re-write this as a loop so it could be used inside kernels?  See https://github.com/oneapi-src/oneAPI-samples/blob/master/DirectProgramming/DPC%2B%2BFPGA/ReferenceDesigns/mvdr_beamforming/src/StreamingQRD.hpp

Reply:
Done.